python/lsst/meas/algorithms/matcherSourceSelector.py

```diff
@@ -162,4 +162,71 @@ def _isUsable(self, source):
             and not source.get(self.fluxFlagKey) \
             and self._goodSN(source)
 
+
+class MatcherPessimisticSourceSelectorTask(MatcherSourceSelectorTask):
+    """
+    !Select sources that are useful for matching.
+
+    Good matching sources have high signal/noise, are non-blended. They need not
+    be PSF sources, just have reliable centroids.
+    """
+    def _getSchemaKeys(self, schema):
+        """Extract and save the necessary keys from schema with asKey."""
+        self.parentKey = schema["parent"].asKey()
+        self.centroidXKey = schema["slot_Centroid_x"].asKey()
+        self.centroidYKey = schema["slot_Centroid_y"].asKey()
+        self.centroidFlagKey = schema["slot_Centroid_flag"].asKey()
+
+        fluxPrefix = "slot_%sFlux_" % (self.config.sourceFluxType,)
+        self.fluxField = fluxPrefix + "flux"
+        self.fluxKey = schema[fluxPrefix + "flux"].asKey()
+        self.fluxFlagKey = schema[fluxPrefix + "flag"].asKey()
+        self.fluxSigmaKey = schema[fluxPrefix + "fluxSigma"].asKey()
+
+        self.edgeKey = schema["base_PixelFlags_flag_edge"].asKey()
+        self.interpolatedCenterKey = schema["base_PixelFlags_flag_interpolatedCenter"].asKey()
+        self.saturatedKey = schema["base_PixelFlags_flag_saturated"].asKey()
+
+    def _isUsable_vector(self, sourceCat):
+        """
+        Return True for each source that is usable for matching, even if it may
+        have a poor centroid.
+
+        For a source to be usable it must:
+        - have a valid centroid
+        - not be deblended
+        - have a valid flux (of the type specified in this object's constructor)
+        - have adequate signal-to-noise
+        """
+        return self._hasCentroid_vector(sourceCat) \
+            & self._isParent_vector(sourceCat) \
+            & self._goodSN_vector(sourceCat) \
+            & ~sourceCat.get(self.fluxFlagKey) \
+            & ~sourceCat.get(self.edgeKey) \
+            & ~sourceCat.get(self.interpolatedCenterKey) \
+            & ~sourceCat.get(self.saturatedKey)
+
+    def _isUsable(self, source):
+        """
+        Return True if the source is usable for matching, even if it may have a
+        poor centroid.
+
+        For a source to be usable it must:
+        - have a valid centroid
+        - not be deblended
+        - have a valid flux (of the type specified in this object's constructor)
+        - have adequate signal-to-noise
+        """
+        return self._hasCentroid(source) \
+            and self._isParent(source) \
+            and not source.get(self.fluxFlagKey) \
+            and self._goodSN(source) \
+            and not source.get(self.edgeKey) \
+            and not source.get(self.interpolatedCenterKey) \
+            and not source.get(self.saturatedKey)
+
+
 sourceSelectorRegistry.register("matcher", MatcherSourceSelectorTask)
+sourceSelectorRegistry.register("matcherPessimistic",
+                                MatcherPessimisticSourceSelectorTask)
```

Comment:
If you rename the class (as I hope you will) please also adjust the name you use to register it.